katunold/Fast-Food-react-webpack
# #163754506 Implement sign-up functionality

#### What this PR does?
- Implement sign-up form with live validations for both admin and client.
- Implement a spinner component.
- Add a notification component  
- Write tests.
#### Description of Task to be completed?
- A user should be able to access a well styled up form for signup.
- A user should be able to view error messages in-case invalid data is input.
- Users should be able to Signup and or Register using Sign up forms on the signup route.
#### How should this be manually tested?
After cloning the repo, cd into it and follow the steps 
* In terminal, type `git checkout ft-sig-up-form-163754506` to switch branch.
* Then type `npm install` to install all the project dependencies in package.json **Note** ensure that you have nodejs installed on your computer. 
* Finally, type `node server` to run the project server and then open your browser and type `http://localhost:3000/signup` to access the signup for a `normal user` and `http://localhost:3000/admin` to signup as an admin.
[Link to Demo](https://fast-food-fast-arnold-webpack.herokuapp.com)
#### What are the relevant pivotal tracker stories?
[#163754506](https://www.pivotaltracker.com/story/show/163754506)
#### Screenshots
**Empty sign-up form normal user**
![image](https://user-images.githubusercontent.com/25249904/52465321-49346900-2b8f-11e9-85ba-ae60b6058ee7.png)
**Empty sign-up admin form**
![image](https://user-images.githubusercontent.com/25249904/52465362-6f5a0900-2b8f-11e9-8fbb-cc00d2c43b20.png)
**Form input Live Validations**
![image](https://user-images.githubusercontent.com/25249904/52465434-ab8d6980-2b8f-11e9-94a6-065fc5eca7b1.png)
